recipes/sdl2/all/conanfile.py

```diff
@@ -266,8 +270,10 @@ def _build_cmake(self):
         if self.settings.os == "Linux":
             if self.options.pulse:
                 os.rename("libpulse.pc", "libpulse-simple.pc")
-        cmake = self._configure_cmake()
-        cmake.build()
+        with tools.run_environment(self):
+            with tools.environment_append({"LIBRARY_PATH": os.getenv("LD_LIBRARY_PATH")}):
+                cmake = self._configure_cmake()
+                cmake.build()
```

Comment:
This should work too, but uses semi-deprecated `RunEnvironment`
```suggestion
        with tools.environment_append({"LIBRARY_PATH": RunEnvironment(self).vars["LD_LIBRARY_PATH"]}):
            cmake = self._configure_cmake()
            cmake.build()
```

Reply:
This is probably better
```suggestion
        lib_paths = [lib for dep in self.deps_cpp_info.deps for lib in self.deps_cpp_info[dep].lib_paths]
        with tools.environment_append({"LIBRARY_PATH": os.pathsep.join(lib_paths)}):
            cmake = self._configure_cmake()
            cmake.build()
```